### DISCUSSION TITLE
How to enable RLS in PostgreSQL

### DISCUSSION THREAD

**Discussion by kevinmamaqi** [prisma/prisma#10486] [Q&A] [answered]
Is it possible to enable Postgresql RLS with Prisma? If so, how? - I couldn't find information on it apart from dead links.

Thanks for the fantastic product

**Comment by TasinIshmam** [comment, marked as answer]
Hey @kevinmamaqi 

Unfortunately, this is not something directly supported by Prisma right now. There's a related [issue](https://github.com/prisma/prisma/issues/5128) that you might consider checking out.

**Reply by kevinmamaqi** [reply]
Thank you. Do you know if it is possible to create custom SQL migrations that don't colide with Prisma schema? so I can add RLS and indexes manually to postgres, but still available when executing a migration

**Reply by TasinIshmam** [reply]
Technically yes, but I haven't tried it in the past so I'm not totally sure what the experience will be like. In general though, all the Prisma migration files are _valid SQL_ so you can put any arbitrary SQL in your migration files and it will be added to your migration history. 

This is what you have to do: 

1. Generate a new migration _without applying it to the database_ with the `--create-only` flag. 
```
npx prisma migrate dev --create-only
```

2. Take a look at the generated `migration.sql` file and update it accordingly to add any new SQL statements. 

3. Apply the migrations `npx prisma migrate dev` 


**Important Note**: Always make sure you apply the changes to `migration.sql` _before_ applying it to your database. Modifying an already applied migration file will lead to a corrupt migration history.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
